tests/PHPStan/Analyser/AnalyserIntegrationTest.php

```diff
@@ -1075,6 +1080,24 @@ public function testBug8503(): void
 		$this->assertNoErrors($errors);
 	}
 
+	public function testExplicitNever(): void
+	{
+		$initializerExprTypeResolver = self::getContainer()->getByType(InitializerExprTypeResolver::class);
+
+		$result = $initializerExprTypeResolver->getPlusType(
```

Comment:
I did only a single test as a representative for one operation.

should I test every single one?

Reply:
That's fine.

Reply:
But you should test all the branches obviously, ideally with a dataProvider. So when the left type is explicit, when the right type is explicit, and when none of them are.

Reply:
good idea. done.